### ISSUE TITLE
Deployment documentation

### ISSUE THREAD

**Issue by MFernstrom** [RapidInfoSys/Rapid#2]
I can't find information about deployment paths, JVM tuning, DevOps related tools/paths.

**Comment by gareth-edwards** [comment]
Wow. Could you provide an example of the sort of thing you're expecting?

**Comment by MFernstrom** [comment]
Sure!

I *just* found Rapid, so please excuse me if I'm wrong about any of this.

It looks to me that Rapid is mostly a single-server, work-on-the-server, framework/tool.

I didn't find anything about how the URLs work/are setup (To easily map with Nginx for example).

We all know Java is finicky, would love to see some tuning information to get the most out of each instance/server.

As for DevOps/release paths, I'd love to see something like this:

Developers work on a local copy, exports the new version, enters it into Nexus (Or whatever repo tool)

A deployment server then uploads that exported file to several Rapid servers and sets the new version via APIs.

Doing that allows to a bigger more robust deployment (load balancing, up/down checks, blue/green deployment, auto scale on-demand etc.)

**Comment by gareth-edwards** [comment]
Cool. Thanks Marcus, that's some great stuff there.

Yes, you're right: Rapid is mostly a single-server, work-on-the-server, framework/tool. For now. _BUT_ we do have some deployments that are clustered across 2 or more Tomcat servers with Apache Httpd doing the load balancing. Haven't tried Nginx but I hear good things and just might try it in the future. At the moment it requires manual updates to each member of the cluster (and sticky sessions), but yes, the grand vision is certainly a synchronised multiple deployment via APIs.

For tuning we just crank up the heap to about 80% of available memory. Would be awesome to have some qualitative data on really effective tuning but haven't gotten there yet. Our largest customer has a 100 or so daily users running a reasonably large app with lots of integration on a single (non-clustered) Rapid server, and of course good optimisation would get it even better, but they're happy with the response speed so we've never really looked into it. (Rapid is currently mainly used for local government back office business applications in the UK, call centres, process automation, etc).

But those are some awesome ideas. Thanks so much. Hopefully you'll stay watching and see how Rapid matures.

**Comment by MFernstrom** [comment]
Yeah I'm trying to give it a good solid tryout, if it turns out nice I'm going to try to pitch it to management.

Expect lots of silly questions!    :)
